airflow/config_templates/config.yml

```diff
@@ -2173,6 +2173,13 @@
       type: string
       example: ~
       default: ""
+    - name: pod_mutation_using_k8s_pod
```

Comment:
I think we will need to support both for 1.10.12 since some of the users have already updated their `pod_mutation_hook` to use k8s.Pod.

We could just check the type of the Pod that is passed, example:


```
if isinstance(k8s.V1Pod):
    # Use Pod from k8s api
    ....
else:
    # Use Pod from Airflow POD
```

Reply:
@kaxil I tried to figure that out but... here's where it gets tricky. 

We now need to determine what type to send INTO the pod_mutation_hook function. If a user is using an existing pod_mutation_hook Pod object, then they need to RECIEVE the Pod instead of a Kubernetes V1Pod otherwise the function will break. Does that make sense? If this were a typed language we'd need BOTH of these functions

```
def pod_mutation_hook(pod: k8s.V1Pod) ->  k8s.V1Pod
```

AND 

```
def pod_mutation_hook(pod: airflow.kubernetes.pod.Pod) ->  airflow.kubernetes.pod.Pod
```

I'm not sure how to handle that other than setting a flag.

Reply:
But based on the following can we not send the type that we receive?

```
if isinstance(k8s.V1Pod):
    # Use Pod from k8s API
    pod_mutation_hook(pod)
    ....
else:
    # Use Pod from Airflow POD
    pod_mutation_hook(pod)
```

If it is just for TypeHints, we can avoid typehints as Python2.7 does not work with it or use `Optional[k8s.V1Pod, airflow.kubernetes.pod.Pod]`

Reply:
https://docs.python.org/3/library/typing.html#typing.Union

```
Union type; Union[X, Y] means either X or Y.

To define a union, use e.g. Union[int, str].
```

Reply:
I understand what you mean now, apologies, checking to see if we can avoid the situation at all